main_test.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"testing"
+	"time"
+)
+
+func TestDateParamToEpoch(t *testing.T) {
+
+	timeNow = func() time.Time {
+		//16 Aug 1994 15:30
+		return time.Date(1994, time.August, 16, 15, 30, 0, 100, defaultTimeZone)
+	}
+
+	const shortForm = "15:04 2006-Jan-02"
+
+	var tests = []struct {
+		input  string
+		output string
+	}{
+		{"midnight", "00:00 1994-Aug-17"},
+		{"noon", "12:00 1994-Aug-16"},
+		{"teatime", "16:00 1994-Aug-16"},
+
+		{"noon 08/16/94", "12:00 1994-Aug-16"},
+		{"midnight 20060816", "00:00 2006-Aug-16"},
```

Comment:
I see the inconsistency here with `midnight 20060816` meaning start of day, but without a date (above) it means end of day.